biomage/experiment/download.py

```diff
@@ -0,0 +1,306 @@
+import json
+import os
+from enum import Enum
+from fileinput import filename
+from pathlib import Path
+
+import boto3
+import click
+from botocore.exceptions import ClientError
+
+from ..rds.run import run_rds_command
+from ..utils.constants import (
+    CELLSETS_BUCKET,
+    PROCESSED_FILES_BUCKET,
+    RAW_FILES_BUCKET,
+    SAMPLES_BUCKET,
+)
+from .utils import set_modified_date
+
+output_path = "."
+
+
+s3 = boto3.resource("s3")
+SAMPLES = "samples"
+RAW_FILE = "raw_rds"
+PROCESSED_FILE = "processed_rds"
+CELLSETS = "cellsets"
+
+
+def _download_file(bucket, s3_path, file_path):
+    """
+    Utility to download file
+    """
+    file_path.parent.mkdir(parents=True, exist_ok=True)
+
+    s3_obj = s3.Object(bucket, s3_path)
+    s3_obj.download_file(str(file_path))
+    set_modified_date(file_location=file_path, date=s3_obj.last_modified)
+
+
+def _create_sample_mapping(samples_list, output_path):
```

Comment:
how about renaming `_create_sample_mapping` to `map_sample_name_to_id`? Then you won't need that comment to say what the function does, because it will be obvious from the function name

Reply:
The function not only maps the sample name to id, but also creates the file, that's why it's named like that. What do you think?

Reply:
that's true, it also creates the file. Then maybe `_create_sample_mapping` is ok, in this case I would change the comment to add `and writes them to a file`

Reply:
comment added